blueos_repository/docker/models/auth.py

```diff
@@ -0,0 +1,42 @@
+import datetime
+import dataclasses
+from typing import Optional
+
+
+@dataclasses.dataclass
+class AuthToken:
+    """
+    Data structure to store the authorization token information
+
+    Attributes:
+        token (str): An opaque Bearer token that clients should supply to subsequent requests in the Authorization header.
+        access_token (str): A token used for OAuth 2.0 compatibility. Equivalent to the `token` attribute.
+        expires_in (Optional[int]): The duration in seconds since the token was issued that it will remain valid. Defaults to 60 seconds.
+        issued_at (Optional[datetime.datetime]): The UTC time at which the token was issued, in RFC3339 format.
+        refresh_token (Optional[str]): A token that can be used to obtain new access tokens for different scopes.
+    """
+
+    token: Optional[str] = None
+    access_token: Optional[str] = None
+    expires_in: Optional[int] = 60
+    issued_at: Optional[datetime.datetime] = None
+    refresh_token: Optional[str] = None
+
+
+    def __post_init__(self):
+        # Ensure that at least one of `token` or `access_token` is provided.
+        if not self.token and not self.access_token:
+            raise ValueError("Either 'token' or 'access_token' must be specified.")
+
+        self.issued_at = datetime.datetime.fromisoformat(self.issued_at) if self.issued_at else datetime.datetime.now(datetime.timezone.utc)
```

Comment:
```suggestion
        self.issued_at = datetime.datetime.fromisoformat(self.issued_at or datetime.timezone.utc)
```

Reply:
Actually after testing with a None self.issued_at, I found that the use of the `datetime.datetime.fromisoformat(datetime.timezone.utc)` will raise an error, so I dont think that we can actually just use in this way, because `fromisoformat` only acept str as argument.